### PR TITLE
Add Python Implementation to "Swap" Problem

### PR DESCRIPTION
**Pull request by Prithul123**
_If any of the below don't apply to this Pull Request, mark the checkbox as done._

- [x] I have tested my code.
- [x] I have added my solution according to the steps [here](https://usaco.guide/general/adding-solution#steps).
- [x] I have followed the code conventions mentioned [here](https://usaco.guide/general/adding-solution/#code-conventions).
  - I understand that if it is clear that I have not attempted to follow these conventions, my PR will be closed.
  - If changes are requested, I will re-request a review after addressing them.

### CODE REVIEW

**Review by SansPapyrus683 — CHANGES_REQUESTED** [review]
u sure your code runs?

**Comment by SansPapyrus683** [comment]
time-wise

**Comment by Prithul123** [comment]
image

this solution runs in time and works with 100% accuracy.

**Comment by SansPapyrus683** [comment]
image
this is not the code you put in the pr

**Comment by Prithul123** [comment]
> 

Oops, didn't commit properly for some reason, should be good now.

File: solutions/bronze/usaco-1013.mdx [hunk 1]
```diff
@@ -113,6 +113,53 @@ public class swap {
 ```
 
 </JavaSection>
+<PySection>
+
+```py
+def reverse_segment(cows, start, end):
```
Comment (SansPapyrus683) [inline solutions/bronze/usaco-1013.mdx:119]: type annotations

File: solutions/bronze/usaco-1013.mdx [hunk 2]
```diff
@@ -113,6 +113,53 @@ public class swap {
...
+		reverse_segment(cows, b1 - 1, b2 - 1)
+		if tuple(cows) in visited:
+			break
+		visited.add(tuple(cows))
+	# calculate the length of the cycle
+	cycle_len = len(visited)
+	return cycle_len
+
+
+def apply_swaps(cows, a1, a2, b1, b2, k):
```
Comment (SansPapyrus683) [inline solutions/bronze/usaco-1013.mdx:141]: you only call this once, just put it in the main code

File: solutions/bronze/usaco-1013.mdx [hunk 3]
```diff
@@ -113,6 +113,53 @@ public class swap {
...
+
+n, k = map(int, input().split())
+a1, a2 = map(int, input().split())
+b1, b2 = map(int, input().split())
+
+cows = list(range(1, n + 1))
+cows = apply_swaps(cows, a1, a2, b1, b2, k)
+
+for cow in cows:
+	print(cow)
```
Comment (SansPapyrus683) [inline solutions/bronze/usaco-1013.mdx:159]: wasn't this when you still had to print to a file?

**Review by SansPapyrus683 — CHANGES_REQUESTED** [review]

File: solutions/bronze/usaco-1013.mdx [hunk 4]
```diff
@@ -116,47 +116,51 @@ public class swap {
...
-def apply_swaps(cows, a1, a2, b1, b2, k):
-	cycle_len = find_cycle_length(cows, a1, a2, b1, b2)
-	k %= cycle_len
-	for i in range(k):
-		reverse_segment(cows, a1 - 1, a2 - 1)
-		reverse_segment(cows, b1 - 1, b2 - 1)
-
-	return cows
-
+def reverse_segment(cows: List[int], start: int, end: int) -> None:
```
Comment (SansPapyrus683) [inline solutions/bronze/usaco-1013.mdx:119]: import list from typing

File: solutions/bronze/usaco-1013.mdx [hunk 5]
```diff
@@ -162,3 +165,61 @@ int main() {
...
+
+
+def reverse_segment(cows: List[int], start: int, end: int):
+	while start < end:
+		cows[start], cows[end] = cows[end], cows[start]
+		start += 1
+		end -= 1
+
+
+def find_cycle_length(cows: List[int], a1: int, a2: int, b1: int, b2: int) -> int:
```
Comment (SansPapyrus683) [inline solutions/bronze/usaco-1013.mdx:183]: this function isn't even used what

Comment (Prithul123) [reply solutions/bronze/usaco-1013.mdx]: oops, that was from another approach. I got rid of it .

**Comment by Prithul123** [comment]
I fixed the issues.

File: solutions/bronze/usaco-1013.mdx [hunk 6]
```diff
@@ -162,3 +165,46 @@ int main() {
...
+while True:
+	reverse_segment(cows, a1 - 1, a2 - 1)
+	reverse_segment(cows, b1 - 1, b2 - 1)
+	if tuple(cows) in visited:
+		break
+	visited.add(tuple(cows))
+
+cycle_len = len(visited)
+k %= cycle_len
+for i in range(k):
```
Comment (SansPapyrus683) [inline solutions/bronze/usaco-1013.mdx:201]:
```suggestion
for _ in range(k):
```

File: solutions/bronze/usaco-1013.mdx [hunk 7]
```diff
@@ -162,3 +165,46 @@ int main() {
...
+	reverse_segment(cows, a1 - 1, a2 - 1)
+	reverse_segment(cows, b1 - 1, b2 - 1)
+
+# print cows
```
Comment (SansPapyrus683) [inline solutions/bronze/usaco-1013.mdx:205]:
```suggestion
```

**Review by SansPapyrus683 — CHANGES_REQUESTED** [review]

File: solutions/bronze/usaco-1013.mdx (hunk 3 above)
Comment (SansPapyrus683) [reply solutions/bronze/usaco-1013.mdx]: bump

Comment (Prithul123) [reply solutions/bronze/usaco-1013.mdx]: bump?

Comment (SansPapyrus683) [reply solutions/bronze/usaco-1013.mdx]: not addressed

Comment (Prithul123) [reply solutions/bronze/usaco-1013.mdx]: what does bump mean?

Comment (SansPapyrus683) [reply solutions/bronze/usaco-1013.mdx]: "bump" this up your priority list or something of the sort
not addressed, still

Comment (Prithul123) [reply solutions/bronze/usaco-1013.mdx]: I'm still not sure what you are asking for, when you say "wasn't this when you still had to print to a file?" what are you asking me to do in the code?

Comment (SansPapyrus683) [reply solutions/bronze/usaco-1013.mdx]: your code uses stdin/stdout
shouldn't this problem require fin/fout?

Comment (Prithul123) [reply solutions/bronze/usaco-1013.mdx]: um I don't use stdin or stdout anywhere in my code

Comment (SansPapyrus683) [reply solutions/bronze/usaco-1013.mdx]: you do? you read from input() & print w/ print()

Comment (Prithul123) [reply solutions/bronze/usaco-1013.mdx]: um is my last commit fine?

**Review by SansPapyrus683 — APPROVED** [review]

**Comment by Prithul123** [comment]
could you merge this pull request because I want to fill out the usaco google form today?

**Review by envyaims — APPROVED** [review]
lgtm

**Pull request merged**